qcodes/instrument_drivers/QuTech/IVVI.py

```diff
@@ -183,7 +183,45 @@ def _get_version(self):
             mes = self.ask(bytes([3, 4]))
             ver = mes[2]
             return ver
-
+    
+    def linsweep(self,start,end,points):
+        """
+        A replacement of numpy.linspace for use with the ivvi rack dacs.
+        ivvi.linsweep gives:
+         - even spacing in voltages
+         - voltages match the set voltages
+         - protection from oversampling
+         
+        Args:
+            start (number): starting voltage
+            end (number): ending voltage
+            points (int): number of points between start and end, inclusive
+            
+        Returns:
+            list of valid voltages for the ivvi rack
+            
+        Examples:
+            #normal dac aligned voltages
+            ivvi.linsweep(-200,200,201) -> [ -199.404, -197.391 ... ] 
+            
+            #protection from oversampling
+            ivvi.linsweep(start=108.8,end=109,points=11) -> 
+                [108.82734416723888, 108.88838025482566,108.94941634241245, 109.01045242999923]
+        """
+        points = max(round(abs(points)),2)
+        use_reversed = end < start
+        if use_reversed:
+            start,end = end,start
```

Comment:
why is "automatic fixing" favored against over "strictness"? i'd suggest to raise an exception.

either way, this behavior should be documented in the docstring.

Reply:
what is your opinion on this comment?

Reply:
This is not a case of automatic fixing. If the user wants decreasing values, they get it. 
It should be possible to the the algorithm generate values similar to numpy.linspace. 
The  algorithm only works forwards (increasing values), so we detect that case and at the end reverse the output to match what is expected.

Reply:
ok, justified. but the `this behavior should be documented in the docstring` comment still holds :)